# 8.1

## a)

In [1]:
import numpy as np
import scipy.stats as st

alpha = 0.05
median = 70 # null hypothesis
sample = np.array([71, 69, 67, 68, 73, 72, 71, 71, 68, 72, 69, 72])
n_positives = sample[sample-median > 0].size
n_total = sample.size
p = st.binom_test(x=n_positives, n=n_total, p=0.5, alternative="less")
print('p={:.3f}'.format(p))

p=0.806


Mit einem Signifikanzniveau von $5\%$ kann die Nullhypothese bei einem p-Wert von $0.806$ nicht verworfen werden. Der Weinhändler betrügt wahrscheinlich nicht.

## b)

In [14]:
diff = sample - median
p = st.wilcoxon(x=diff, correction=True)[1]
print('p={:.3f}'.format(p))

p=0.690


Beim Wilcoxon-Test fällt der p-Wert mit $0.69$ zwar geringer aus, es reicht aber dennoch nicht, um die Nullhypothese bei einem Signifikanzniveau von $5\%$ zu verwerfen.

# 8.2

## a)

- gepaarte Stichprobe (nicht unabhängig)
- einseitiger Test (nur die _vermehrte_ Anhäufung ist von Interesse)
- $H_0$: Das Rauchen einer Zigarette führt _nicht_ zur Anhäufung von Blutplättchen.
- $H_A$: Das Rauchen einer Zigarette führt zu einer Anhäufung von Blutplättchen.

## b)

- ungepaarte Stichprobe (unabhängig), es wird ein Herstellungsverfahren (im biologischen Sinn) überprüft
    - TODO: In der Löung steht _gepaarte Stichprobe_, hier wird aber nicht die Ausgangspflanze mit der neuen Pflanze verglichen (Vergleich zweier Generationen), sondern es werden jeweils die Pflanzen der neuen Generation miteinander verglichen!
- einseitiger Test (es wird auf _mehr_ Lebenskraft bzw. auf eine grössere Höhe geprüft)
- $H_0$: Die selbst- und fremdbefruchteten Setzlinge sind im Mittel gleich gross.
- $H_A$: Die fremdbefruchteten Setzlinge werden im Mittel grösser als die selbstbefruchteten.

## c)

- ungepaarte Stichprobe (unabhängig)
- zweiseitiger Test (es wird geprüft, _ob_ eine Veränderung eintritt, nicht, ob diese Änderung positiv oder negativ ist)
- $H_0$: Das Kalziumgehalt in der Nahrung beeinflusst den systolischen Blutdruck nicht.
- $H_A$: Das Kalziumgehalt in der Nahrung beeinflusst den systolischen Blutdruck.

## d)

- ungepaarte Stichprobe (unabhängig)
- zweiseitiger Test (es wird geprüft, ob die beiden Formen von Eisen _unterschiedlich_ gut aufgenommen werden; dies kann besser/mehr oder schlechter/weniger sein)
- $H_0$: Die beiden Formen von Eisen $Fe^{2+}$ und $Fe^{3+}$ werden gleich gut aufgenommen.
- $H_A$: Die beiden Formen von Eisen $Fe^{2+}$ und $Fe^{3+}$ werden _nicht_ gleich gut aufgenommen.

# 8.3

## a)

Es handelt sich um gepaarte Stichproblem: Es wird jeweils das Gleiche auf unterschiedliche Art gemessen.

## b)

Modell:

$$ D_1, ... , D_n \stackrel{i.i.d}{\sim} \mathcal{N}(\mu, \sigma^2) $$

$ \sigma $ durch $ \hat{\sigma} $ geschätzt

Nullhypothese: Die beiden Messgeräte produzieren keine systematisch unterschiedlichen Messwerte. (Schwankungen sind reiner Zufall.)

$$ H_0: \mu_D = \mu_0 = 0 $$

Alternativhypothese: Gerät B misst systematisch grössere Werte:

$$ H_A: \mu_D < \mu_0 $$

Teststatistik: $\mathcal{t}$-Verteilung mit Freiheitsgrad $8$

Verteilung under $H_0$:

$$ D \sim \mathcal{N}(0, \sigma_{D}^2) $$

Signifikanzniveau:

$$ \alpha=0.05 $$

In [33]:
import numpy as np
from scipy.stats import t

alpha = 0.05
sample_a = np.array([120, 265, 157, 187, 219, 288, 156, 205, 163])
sample_b = np.array([127, 281, 160, 185, 220, 298, 167, 203, 171])
diff = sample_b - sample_a
n = diff.size
mean = diff.mean()
std = diff.std(ddof=1)

lower = t.ppf(q=1-alpha, df=n-1, loc=0, scale=std/np.sqrt(n))
print('[{:.3f},∞)'.format(lower))

[3.843,∞)


Verwerfungsbereich:

_Anmerkung_: Die Lösung sieht den Verwerfungsbereich auf der anderen Seite. (Die Differenz $D$ kann als $B-A$ oder $A-B$ berechnet werden.)

$$ [3.843,\infty) $$

In [37]:
p = 1 - t.cdf(x=diff.mean(), df=n-1, loc=0, scale=std/np.sqrt(n))
print('p={:.3f}'.format(p))
print('μ={:.3f}'.format(mean))

p=0.012
μ=5.778


Testentscheid: $p<\alpha$ und $\mu \in [3.843,\infty)$: Die Nullhypothese wird verworfen!

Interpretation: Messgerät B produziert mit einer hohen Wahrscheinlichkeit grössere Messwerte.

## c)

$$ Z \sim \mathcal{Bin}(9,0.5) $$

$Z$ ist binomialverteilt: das gleiche Modell wie bei einem Münzwurf. Einmal ist der Messwert per Zufall höher, einmal tiefer, aber bei vielen Messungen werden höhere und tiefere Messwerte praktisch gleich oft auftreten.

# 8.4

$$ \overline{x}_n=113.4, \overline{y}_n=106.6, \hat{\sigma}_x^2=13.82, \hat{\sigma}_y^2=5.16 $$

## a)

Es handelt sich um eine _ungepaarte Stichprobe_ (obwohl es sich um Goldschakal-Päärchen handeln könnte), denn es werden verschiedene, unabhängige Prüfobjekte betrachtet.

## b)

$H_0$: Es gibt keinen signifikanten Unterschied zwischen der Kieferlänge männlicher und weiblicher Schakale. $\mu_x=\mu_y$

$H_A$: Es gibt einen signifikanten Unterschied zwischen der Kieferlänge männlicher und weiblicher Schakale $\mu_x \neq \mu_y$

Da die Frage nach einem _Unterschied_ lautet, muss ein zweiseitiger Test angewendet werden.

## c)

In [16]:
import pandas as pd
import scipy.stats as st

jackals = pd.read_csv('jackals.txt')
p = st.ttest_ind(jackals['M'], jackals['W'], equal_var=False)[1]
print('p={:.4f}'.format(p))

p=0.0034


Von einem Signifikanzniveau von $\alpha=0.05$ ausgehend (steht nirgends geschrieben), kann die Nullhypothese verworfen werden. Es gibt höchstwahrscheinlich einen signifikanten Unterschied zwischen der Kieferlängen männlicher und weiblicher Goldschakale.

## d)

In [15]:
p = st.mannwhitneyu(jackals['M'], jackals['W'], alternative="two-sided")[1]
print('p={:.4f}'.format(p))

p=0.0048


Der p-Wert liegt knapp unter dem Signifikanzniveau von $\alpha=0.05$. Die Nullhypothese kann auch mit dem Mann-Whitney U-Test verworfen werden.

## e)

Da es sich bei den Kieferlängen um ein natürliches Phänomen handelt, dürften diese normalverteilt sein.

Wenn jedoch der eine Test signifikant und der andere nicht signifikant ausgefallen wäre, würde ich weiter Testdaten sammeln, und diese noch einmal mit beiden Testverfahren prüfen.

_Lösung:_ Normalverteilung darf nicht angenommen werden, darum ist der Mann-Whitney-U-Test vertrauenswürdiger. Die grosse Abweichungen der p-Werte sind jedoch problematisch.

# 8.5

In [5]:
import numpy as np

zurich = np.array([16.3, 12.7, 14.0, 53.3, 117, 62.6, 27.6])
basel = np.array([10.4, 8.91, 11.7, 29.9, 46.3, 25.0, 29.4])

## a)

In [6]:
diff = zurich - basel
diff_mean = diff.mean()
diff_std = diff.std(ddof=1)
print('μ={:.3f}, σ={:.3f}'.format(diff_mean, diff_std))

μ=20.270, σ=26.272


## b)

Es handelt sich um eine ungepaarte Stichprobe, denn es werden _unterschiedliche Sachen_ gemessen, nicht das gleiche zweimal.

## c)

$$H_0: \mu_d = 0$$
$$H_A: \mu_d \neq 0$$

## d)

Teststatistik: Die Differenzen der beiden Messreihen, $D \sim \mathcal{N}$

In [7]:
import scipy.stats as st

p = st.ttest_ind(zurich, basel, equal_var=False)[1]
print('p={:.4f}'.format(p))

p=0.2233


## e)

In [8]:
p = st.mannwhitneyu(zurich, basel, alternative='greater')[1]
print('p={:.4f}'.format(p))

p=0.1251
